### ISSUE TITLE
segmentation fault issue depending on hyperparameter

### ISSUE THREAD

**Issue by sbyoo** [lmcinnes/umap#399]
Hello. 

I was facing weird occasions where UMAP works perfectly for a set of hyperparameter while another set of hyperparameter causing error exhibiting 'segmentation fault'. It is not due to data itself nor virtualenv since they were identical. FYI, data is about (500,000 x 39) numpy array. 

The versions of packages are:
- umap-learn == 0.4.0 (but was also same in 0.3.0 and 0.4.0rc3).
- numba ==0.48.0 (I tried 0.40.0, 0.44.0, 0.46.0 but the result was identical)
- scikit-learn ==0.22.0
- numpy==1.18.2
- scipy == 1.4.0
- llvmlite == 0.31.0
- joblib==0.14.1
- python=3.6.1 (I tested with python=3.6.4 also)

The functioning hyperparameter set is:
n_neighbor: 30, 50 (I did not search all yet)
min_dist: 0.005 to 0.5
metric: correlation
dimension:2
The malfunctioning hyperparameter set it 
n_neighbor: 70, 90, 110 (I did not search all yet)
min_dist: 0.005 to 0.5 (these are identical from above)
metric: correlation
dimension:2

If there is any diagnostic aspects, please let me know. Thank you.

**Comment by lmcinnes** [comment]
I'm guessing that there is an odd corner case bug in the correlation metric -- that would be the most likely case. I'll have a look tomorrow and see if I can find a cause.

**Comment by sbyoo** [comment]
Indeed. I forgot to mention it seemed to work in the Euclidean metric but not in correlation metric. Thank you for the quick response.

**Comment by lmcinnes** [comment]
I'm not seeing anything obvious. Is this a dataset that you are able to share? (I certainly understand that this is often not the case). If so I can try to reproduce this locally and figure out what the issue might be.

**Comment by sbyoo** [comment]
@lmcinnes. 
Thank you for quickly reaching me after testing it. 
Three things to report: 1. In the same setup, if I reduce the size of the data and run on the malfunctioning hyperparameter set, it worked. 
2. I created totally new environment in conda and installed python 3.6.5 and some of the hyperparameters that did not work before are working (not all of them). Difference other than python version is scipy == 1.4.1 and scikit-learn==0.22.1. 
3. Sorry that I could not try with my old mac by the error message 'low memory' (it has 8GB RAM).

For the data, I will have to ask my PI whether it is fine to share data for this purpose (I think he will be positive). I will send you an email on how to share the data.
